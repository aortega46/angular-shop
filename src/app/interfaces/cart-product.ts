export interface CartProduct {
  id: number
  title: string
  price: number
  thumbnail: string
  quantity: number
}
