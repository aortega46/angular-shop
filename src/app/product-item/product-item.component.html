<li
  class="flex h-full flex-col justify-between gap-4 rounded bg-[#111] p-4 text-white shadow">
  <img
    class="aspect-video w-full rounded object-cover"
    src="{{ product.thumbnail }}"
    alt="{{ product.title }}" />
  <div class="flex justify-center">
    <p>
      <strong>{{ product.title }}</strong> | ${{ product.price }}
    </p>
  </div>
  <div class="flex items-center justify-center">
    <button
      (click)="isOnCart ? removeFromCart() : addToCart()"
      class="flex items-center justify-center rounded px-4 py-2 {{
        isOnCart ? 'bg-red-400' : 'bg-blue-400'
      }} ">
      <mat-icon class="material-icons-outlined">
        {{ isOnCart ? 'remove_shopping_cart' : 'add_shopping_cart' }}
      </mat-icon>
    </button>
  </div>
</li>
