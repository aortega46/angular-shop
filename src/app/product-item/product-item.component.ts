import { Component, EventEmitter, Input, Output } from '@angular/core'
import { MatIconModule } from '@angular/material/icon'
import { Product } from '../interfaces/product'

@Component({
  selector: 'product-item',
  standalone: true,
  imports: [MatIconModule],
  templateUrl: './product-item.component.html',
})
export class ProductItemComponent {
  // * At this point,signal inputs exists so it can be done with that
  // https://angular.io/guide/signal-inputs#why-should-we-use-signal-inputs-and-not-input
  @Input() product!: Product

  @Output() onAddToCart = new EventEmitter<Product>()

  addToCart() {
    this.onAddToCart.emit(this.product)
  }
}
