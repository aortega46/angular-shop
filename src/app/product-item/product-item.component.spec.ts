import { ComponentFixture, TestBed } from '@angular/core/testing'

import { ProductItemComponent } from './product-item.component'
import { Product } from '../interfaces/product'

const productMock: Product = {
  id: 1,
  title: 'Test',
  price: 100,
  thumbnail: 'test.png',
}

describe('ProductItemComponent', () => {
  let component: ProductItemComponent
  let fixture: ComponentFixture<ProductItemComponent>

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ProductItemComponent],
    }).compileComponents()

    fixture = TestBed.createComponent(ProductItemComponent)
    component = fixture.componentInstance

    component.product = productMock

    fixture.detectChanges()
  })

  it('should create', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    const compiled = fixture.nativeElement
    expect(compiled).toMatchSnapshot()
  })
})
