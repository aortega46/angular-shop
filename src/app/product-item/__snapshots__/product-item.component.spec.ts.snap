// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItemComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <li
    class="flex h-full flex-col justify-between gap-4 rounded bg-[#111] p-4 text-white shadow"
  >
    <img
      alt="Test"
      class="aspect-video w-full rounded object-cover"
      src="test.png"
    />
    <div
      class="flex justify-center"
    >
      <p>
        <strong>
          Test
        </strong>
         | $100 
      </p>
    </div>
    <div
      class="flex items-center justify-center"
    >
      <button
        class="bg-blue-400 flex items-center justify-center px-4 py-2 rounded"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons-outlined material-icons mat-ligature-font mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
           add_shopping_cart 
        </mat-icon>
      </button>
    </div>
  </li>
</div>
`;
